app/javascript/schools/courses/types/curriculum_editor/CurriculumEditor__QuizQuestion.re

```diff
@@ -15,41 +15,37 @@ let answerOptions = t => t.answerOptions;
 let empty = id => {
   id,
   question: "",
-  answerOptions: [
+  answerOptions: [|
     CurriculumEditor__AnswerOption.empty("0", true),
     CurriculumEditor__AnswerOption.empty("1", false),
-  ],
+  |],
 };
 
-let decode = json =>
-  Json.Decode.{
-    id: json |> field("id", string),
-    question: json |> field("question", string),
-    answerOptions:
-      json
-      |> field("answerOptions", list(CurriculumEditor__AnswerOption.decode)),
-  };
-
 let updateQuestion = (question, t) => {...t, question};
 
 let newAnswerOption = (id, t) => {
   let answerOption = CurriculumEditor__AnswerOption.empty(id, false);
   let newAnswerOptions =
-    t.answerOptions |> List.rev |> List.append([answerOption]) |> List.rev;
+    t.answerOptions
+    |> Array.to_list
+    |> List.rev
+    |> List.append([answerOption])
+    |> List.rev
+    |> Array.of_list;
```

Comment:
This seems to be adding a new answer option to the front of the array. Why not use standard JS methods? Here's an example that could be added to `ArrayUtils`:

```reason
let copyAndUnshift = (e, t) => {
  let cp = t |> Array.copy;
  cp |> Js.Array.unshift(e) |> ignore;
  cp;
};
```